File: Semestru 4/Medii de proiectare si programare/Laboratoare/Basket/Basket/repositoryes/IRepository.cs
using Baschet.Business;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Basket.repositoryes
{
    public class RepositoryException : ApplicationException
    {
        public RepositoryException() { }
        public RepositoryException(String mess) : base(mess) { }
        public RepositoryException(String mess, Exception e) : base(mess, e) { }
    }

    interface IRepository<E>
    {
        bool save(E e);

        E find(Entity id);

        bool delete(Entity id);

        List<E> getAll();
    }
}

File: Semestru 4/Medii de proiectare si programare/Laboratoare/BaschetGUI/BaschetRepository/src/main/java/repositories/IRepository.java
package repositories;

import models.Entity;

import java.util.ArrayList;

public interface IRepository<E> {
    public boolean save(E e);

    public boolean delete(Entity e);

    public boolean update(Entity e, E newE);

    public E find(Entity e);

    public ArrayList<E> getAll();
}
